hypyp/prep.py

```diff
@@ -66,68 +76,71 @@ def ICA_choice_comp(icas: list, epochs: list) -> list:
-    comp_number = input("Which IC do you want to use as a template?"
+    subject_id = input("Which participant ICA do you want"
+                        " to use as a template for artifact rejection?"
+                        " Index begins at zero. (If you do not want to apply"
+                        " ICA on your data, do not enter nothing and press enter.)")
+
+    component_id = input("Which IC do you want to use as a template?"
                         " Index begins at zero. (If you did not choose"
-                        "a participant number at first question,"
-                        "then do not enter nothing and press enter again"
-                        "to not apply ICA on your data)")
-
-    # applying ICA
-    if (len(subj_numb) != 0 and len(comp_number) != 0):
-        cleaned_epochs_ICA = ICA_apply(icas,
-                                       int(subj_numb),
-                                       int(comp_number),
-                                       epochs)
-    else:
-        cleaned_epochs_ICA = epochs
+                        " a participant number at first question,"
+                        " then do not enter nothing and press enter again"
+                        " to not apply ICA on your data)")
 
-    return cleaned_epochs_ICA
+    if (len(subject_id) == 0 or len(component_id) == 0):
+        return epochs
+
+    return ICA_apply(icas, int(subject_id), int(component_id), epochs)
 
 
-def ICA_apply(icas: int, subj_number: int, comp_number: int, epochs: list) -> list:
+def ICA_apply(icas: List[ICA], subject_id: int, component_id: int, epochs: List[mne.Epochs], plot: bool = True) -> List[mne.Epochs]:
     """
     Applies ICA with template model from 1 participant in the dyad.
     See ICA_choice_comp for a detailed description of the parameters and output.
     """
 
-    cleaned_epochs_ICA = []
+    cleaned_epochs_ICA: List[ICA] = []
+
     # selecting which ICs corresponding to the template
-    template_eog_component = icas[subj_number].get_components()[:, comp_number]
+    template_eog_component = icas[subject_id].get_components()[:, component_id]
 
     # applying corrmap with at least 1 component detected for each subj
-    fig_template, fig_detected = corrmap(icas,
-                                         template=template_eog_component,
-                                         threshold=0.9,
-                                         label='blink',
-                                         ch_type='eeg')
+    corrmap(icas,
+        template=template_eog_component,
+        threshold=0.9,
+        label='blink',
+        ch_type='eeg',
+        plot=plot,
+    )
 
     # labeling the ICs that capture blink artifacts
     print([ica.labels_ for ica in icas])
 
     # selecting ICA components after viz
-    for i in icas:
-        i.exclude = i.labels_['blink']
+    for ica in icas:
+        ica.exclude = ica.labels_['blink']
 
-    epoch_all_ch = []
```

Comment:
Unnecessary array, we don't use it afterwards. We just use a variable in the loop scope below for the epochs of a subject, then add it to the returned list